### PR TITLE
Feat: Start using docker compose V2

### PR DESCRIPTION
**Pull request by Meldiron**
## What does this PR do?

`docker-compose` seems to not have some features Appwrite started using on development. As a first step before implementing these in production is to make Appwrite ready to use new `docker compose` syntax.

This PR removes `docker-compose` installation from  the Appwrite Docker image, and adds `docker compose`.

**This PR does not include documentation changes. They will be required before releasing.**

## Test Plan

(Manual QA)

- Manually build Appwrite image `docker build . -t appwritest`
- Create Appwrite instance using new image:
```sh
docker run -it --rm \
    --volume /var/run/docker.sock:/var/run/docker.sock \
    --volume "$(pwd)"/appwrite:/usr/src/code/appwrite:rw \
    --entrypoint="install" \
    appwritest:latest
```
- Appwrite was started, and working properly

CleanShot 2022-06-09 at 13 40 50

## Related PRs and Issues

x

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

Yes

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -232,12 +233,16 @@ RUN \
...
   docker-cli \
-  docker-compose \
   libgomp \
   && docker-php-ext-install sockets opcache pdo_mysql \
   && apk del .deps \
   && rm -rf /var/cache/apk/*
 
+RUN \
+  mkdir -p $DOCKER_CONFIG/cli-plugins \
+  && curl -SL https://github.com/docker/compose/releases/download/v2.5.0/docker-compose-linux-x86_64 -o $DOCKER_CONFIG/cli-plugins/docker-compose \
```
Comment (TorstenDittmann) [inline Dockerfile:243]: Can we make the version number here as a variable on top?

**Review by TorstenDittmann — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (stnguyen90) [reply Dockerfile:243]: Does this also need to consider architecture as well instead of hardcoding `x86_64`?

Comment (Meldiron) [reply Dockerfile:243]: - Will make version ENV variable
- Don't think we need to consider architecture, right? It is all running in the same environment, inside a docker container. I will add it tho. Won't hurt, may help.

Comment (Meldiron) [reply Dockerfile]: I was not able to use the command suggested by @stnguyen90 , as it is a `bash` script, but Apwrite runs on `sh` (`/bin/sh: /bin/bash: not found`)

I replaced it with just `$(uname -m)` instead, WDYT?

Comment (Meldiron) [reply Dockerfile]: Nevermind, I think I figured it out. I used:

```sh
ARCH=$(uname -m) && if [ $ARCH == "armv7l" ]; then $ARCH="armv7"; fi
```

Comment (stnguyen90) [reply Dockerfile]: > Don't think we need to consider architecture, right? It is all running in the same environment, inside a docker container. I will add it tho. Won't hurt, may help.

I think there are different images for different architectures and the matching image is used automatically. On my rpi:

```shell
$ uname -m
aarch64

$ docker run --rm -it appwrite/appwrite uname -m
aarch64
```

On my (intel) macbook:

```shell
$ uname -m                                      
x86_64

$ docker run --rm -it appwrite/appwrite uname -m
x86_64
```

**Review by TorstenDittmann — APPROVED** [review]

**Pull request merged**